app/controllers/provider_foreman_controller.rb

```diff
@@ -455,6 +437,10 @@ def change_tab
 
   private ###########
 
+  def provider_class_from_provtype
```

Comment:
do not do now...

in the future, would you pass in `params[:provtype]` instead of accessing `params` directly from helper functions?